### PR TITLE
fix(codedeploy): change the load balancer API in server Deployment Gr…

### LINKED ISSUE #2449: Get rid of aws-codedeploy-api and asCodeDeployLoadBalancer
I don't see a problem for the aws-codedeploy module to take a dependency on aws-elbv2 and aws-elb, eliminating the need for this intricate type hierarchy.

### PR DESCRIPTION
**Pull request by skinny85**
…oup.

BREAKING CHANGE: the type of the `loadBalancer` property in ServerDeploymentGroupProps has been changed.

Fixes #2449

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [x] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-codedeploy/README.md [hunk 1]
```diff
@@ -90,21 +90,23 @@ const deploymentGroup = codedeploy.ServerDeploymentGroup.import(this, 'ExistingC
...
 });
 elb.addTarget(/* ... */);
 elb.addListener({
-    // ...
+  // ...
 });
 
 const deploymentGroup = new codedeploy.ServerDeploymentGroup(this, 'DeploymentGroup', {
-    loadBalancer: elb,
+  loadBalancer: codedeploy.LoadBalancer(elb),
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-codedeploy/README.md]: Can this code sample be correct?

File: packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  public static classic(loadBalancer: elb.LoadBalancer): LoadBalancer {
+    return new ClassicLoadBalancer(loadBalancer);
+  }
+
+  /**
+   * Creates a new CodeDeploy load balancer from an Application Load Balancer Target Group.
+   *
+   * @param albTargetGroup an ALB Target Group
+   */
+  public static alb(albTargetGroup: elbv2.ApplicationTargetGroup): LoadBalancer {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts]: I think I'd prefer a longer name here.

Also, I have no idea how this makes sense, to create an ALB "from" a target group. But oh well I suppose?

File: packages/@aws-cdk/aws-codedeploy/README.md (hunk 1 above)
Comment (eladb) [inline packages/@aws-cdk/aws-codedeploy/README.md]: That looks off

File: packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts (hunk 2 above)
Comment (eladb) [reply packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts]: How about `application`?

File: packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    return new AlbLoadBalancer(albTargetGroup);
+  }
+
+  /**
+   * Creates a new CodeDeploy load balancer from a Network Load Balancer Target Group.
+   *
+   * @param nlbTargetGroup an NLB Target Group
+   */
+  public static nlb(nlbTargetGroup: elbv2.NetworkTargetGroup): LoadBalancer {
```
Comment (eladb) [inline packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts]: network

File: packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    return new NlbLoadBalancer(nlbTargetGroup);
+  }
+
+  public abstract readonly generation: LoadBalancerGeneration;
+  public abstract readonly name: string;
+}
+
+class ClassicLoadBalancer extends LoadBalancer {
```
Comment (eladb) [inline packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts]: You could implement this as inline classes, will be much more concise

Comment (skinny85) [reply packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts]: Done.

File: packages/@aws-cdk/aws-codedeploy/README.md (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-codedeploy/README.md]: I was just testing if you're paying attention. You've passed.

File: packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts (hunk 2 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts]: Changed to `application`.

File: packages/@aws-cdk/aws-codedeploy/README.md (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-codedeploy/README.md]: Changed.

**Pull request merged**